math/modular_exponentiation.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * @file
+ * @brief C++ Program for Modular Exponentiation Iteratively.
+ * Calculate the value of an integer a raised to an integer exponent b 
+ * under modulo c.
+ * @note The time complexity of this approach is O(log b).
+ *
+ * Example:
+ * (4^3) % 5 (where ^ stands for exponentiation and % for modulo)
+ * (4*4*4) % 5
+ * (4 % 5) * ( (4*4) % 5 )
+ * 4 * (16 % 5)
+ * 4 * 1
+ * 4
+ * We can also verify the result as 4^3 is 64 and 64 modulo 5 is 4
+ */
+
+#include <iostream> /// for io operations
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+/**
+* @brief This function calculates a raised to exponent b 
+* under modulo c using modular exponentiation
+* @param a integer base
+* @param b unsigned integer exponent
+* @param c integer modulo
+* @return a raised to power b modulo c
+*/
+int power(int a, unsigned int b, int c) 
```

Comment:
It is better/recommended to use `uint64_t`, rather than `int`
```suggestion
uint64_t power(uint64_t a, unsigned uint64_t b, uint64_t c) 
```

or even better:

```suggestion
template <typename T>
T power(T a, unsigned T b, T c)
```